### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by looh16**
**Add classes:**
1. Classroom
2. Book
3. Rental

 _**Add associations between classes**_

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -4,6 +4,9 @@ class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
+  has_many :rentals
+  has_many :books, through: rentals
+
```
Comment (sja-thedude) [inline person.rb:9]: - [ ] Kindly make sure that after initialize method , a method for adding rentals should be present

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  belongs_to :student
+  belongs_to :book
+
+  attr_accessor :date
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
```
Comment (sja-thedude) [inline rental.rb]: - [ ] Kindly make sure that the rental is adding itself to the @book achieving the 'belongs-to' association. Example :
```ruby    
book.rentals << self
```
Also, you have to do the same for the @person.

File: student.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
 class Student < Person
   attr_accessor :classroom
 
+  belongs_to :classroom
+
```
Comment (sja-thedude) [inline student.rb:3]: - [ ] Kindly add a method that serves as the 'belongs-to' side (a student belongs to a classroom).

File: classroom.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Classroom
+  has_many :students
+  attr_accessor :label
+
+  def initialize(label)
+    @label = label
+  end
```
Comment (sja-thedude) [inline classroom.rb:8]: - [ ] Kindly add a method that serves as the 'has-many' side (a classroom has many students).

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
+
+  has_many :rentals
+  has_many :persons, through: rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
```
Comment (sja-thedude) [inline book.rb:10]: - [ ] Kindly make sure to initialize @rentals in the initialize method.

File: book.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  end
+end
```
Comment (sja-thedude) [inline book.rb:17]: - [ ] Kindly add a method for adding rentals should be present.

File: rental.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  belongs_to :student
+  belongs_to :book
+
+  attr_accessor :date
```
Comment (sja-thedude) [inline rental.rb]: - [ ] Make sure Getters and Setters for @book and @person instance variables should be present.

File: person.rb [hunk 8]
```diff
@@ -4,6 +4,9 @@ class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
```
Comment (sja-thedude) [inline person.rb]: - [ ] Make sure that Getters and Setters for @rentals should be present.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @looh16, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Correct Gitflow
✔️ Linters Passing

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @looh16, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by looh16** [comment]
Thanks!

**Pull request merged**